cli/src/main/java/org/eclipse/hono/cli/AbstractCliClient.java

```diff
@@ -44,32 +54,91 @@
      */
     protected Context ctx;
     /**
-     * The Spring Boot profiles that are active.
+     * The class containing all client properties.
      */
-    protected List<String> activeProfiles;
-
+    protected ClientConfigProperties honoClientConfig;
+    /**
+     * To connection created.
+     */
+    protected HonoConnection con;
+    /**
+     * To stop the executions of internal commands.
+     */
+    protected CountDownLatch connLatch;
     /**
-     * Sets the Spring environment.
-     * 
-     * @param env The environment.
-     * @throws NullPointerException if environment is {@code null}.
+     * Spring Shell inputReader.
      */
     @Autowired
-    public final void setActiveProfiles(final Environment env) {
-        Objects.requireNonNull(env);
-        activeProfiles = Arrays.asList(env.getActiveProfiles());
+    protected @Lazy InputReader inputReader;
+    /**
+     * Spring Shell output helper.
+     * <p>
+     * it can be passed to the method class instead to use the log (it's more stylish).
+     */
+    @Autowired
+    protected ShellHelper shellHelper;
+    /**
+     * The client factories methods.
+     */
+    protected ConnectionLifecycle<HonoConnection> clientFactory;
+    /**
+     * Sets the default client configuration.
+     *
+     * @param honoClientConfig The Client Configuration class instance.
+     */
+    @Autowired
+    private void setClientConfigProperties(final ClientConfigProperties honoClientConfig) {
+        this.honoClientConfig = Objects.requireNonNull(honoClientConfig);
     }
-
     /**
      * Sets the vert.x instance.
-     * 
+     *
      * @param vertx The vert.x instance.
      * @throws NullPointerException if vert.x is {@code null}.
      */
     @Autowired
-    public final void setVertx(final Vertx vertx) {
+    protected void setVertx(final Vertx vertx) {
         this.vertx = Objects.requireNonNull(vertx);
         this.ctx = vertx.getOrCreateContext();
     }
+    /**
+     * Handler for the Hono connection establishment.
+     *
+     * @param startup The async result of the connection.
+     */
+    protected void handleConnectionStatus(final AsyncResult<HonoConnection> startup) {
+        if (startup.succeeded()) {
+            con = startup.result();
+            shellHelper.printInfo("Connection created!");
+        } else {
+            shellHelper.printError("Error occurred during initialization of receiver: " + startup.cause().getMessage());
+            if (connLatch != null) {
+                connLatch.countDown();
+            }
+        }
+    }
+
+    /**
+     * Check the availability of the factory.
+     * If not present it doesn't allow the execution of commands.
+     *
+     * @return Availability status.
+     */
+    protected Availability connectionAvailability() {
```

Comment:
If I am not mistaken, then this method will operate on the *clientFactory* field defined by this abstract class. Thus, invoking it from any of its subclasses will not yield the desired effect of checking the status of the factory kept in the *clientFactory* fields defined in those subclasses.